### DISCUSSION TITLE
Is it possible for the user to progress the stream?

### DISCUSSION THREAD

**Discussion by mustafaalperendogan** [elsa-workflows/elsa-core#1173] [Q&A]
Can we add user objects to the stream?  Can the user direct the flow using forward or cancel buttons?

**Comment by sfmskywalker** [comment]
Hi!

Can you give a little bit more detail about this scenario where the user directs the flow using forward or cancel buttons? Who is the "user" in this scenario? The workflow developer, or a user of your application that is driven by the workflow?
What do you mean with "stream"?

**Reply by mustafaalperendogan** [reply]
First of all thanks for your answer. :)  I want to explain through example.  I'm a developer and I'm going to draw a flow.  I will have 2 objects.  Let these objects be called positions.  1. The position will have actions, forward and cancel.  Your 2nd position will be actions, confirm and cancel.  I will design a view in net core (as a developer).  Position 1 will write to this view and click the forward action.  This view will go to position 2.  The 2nd position will click the confirmation action and the flow will be completed.  In this way, I will receive feedback from each other such as approval or cancellation for a document among my registered users.  I hope I could explain.

**Reply by sfmskywalker** [reply]
That's definitely helpful, thank you.

To clarify, are these "positions" activities on the workflow that you are drawing?
If so, do you want to display the correct view depending on the current activity in the workflow, or the other way around: when the user clicks a button (e.g. Forward or Cancel), it triggers the workflow, which will then move to the next state (e.g. Position 2 when Forward was clicked)?

In any case, it sounds to me all of this is doable.

**Reply by mustafaalperendogan** [reply]
Thank you.
Positions is actually an icon (object) that we add to the stream.  Actions will be added to the features of the position.  View will depend on workflow, workflow will depend on view.  I actually have a class that I wrote myself.  But I want to add objects by drag-and-drop and connect them with lines.  Since I'm just writing it as code it stays very abstract and exceptions happen at runtime.
